--- ./dclib/Makefile.am.org	2006-11-07 02:42:12.000000000 +0000
+++ ./dclib/Makefile.am	2007-08-09 11:00:23.000000000 +0000
@@ -30,5 +30,5 @@
 csearchsocket.h
 
 install-data-local:
-	$(mkinstalldirs) $(datadir)/dclib
-	$(mkinstalldirs) $(datadir)/dclib/plugin
+	$(mkinstalldirs) $(DESTDIR)$(datadir)/dclib
+	$(mkinstalldirs) $(DESTDIR)$(datadir)/dclib/plugin
--- ./Makefile.am.org	2006-11-07 02:42:12.000000000 +0000
+++ ./Makefile.am	2007-08-09 10:59:37.000000000 +0000
@@ -16,8 +16,8 @@
 MAINTAINERCLEANFILES = subdirs configure.in acinclude.m4 configure.files 
 
 install-data-local:
-	$(mkinstalldirs) $(libdir)/pkgconfig
-	$(INSTALL_DATA) dclib.pc $(libdir)/pkgconfig
+	$(mkinstalldirs) $(DESTDIR)$(libdir)/pkgconfig
+	$(INSTALL_DATA) dclib.pc $(DESTDIR)$(libdir)/pkgconfig
 
 package-messages:
 	$(MAKE) -f admin/Makefile.common package-messages
